### PR TITLE
create emit before we use it; make move accept an event.

### LINKED ISSUE #83: Error multitouch emit undefined inside move function
When using a device with multitouch I get an error when outside the chart something is pressed first and held down and afterwards pressing inside the chart.
The error is as following:
TypeError: Cannot read property 'brush' of undefined
TypeError: emitter(...).moved is not a function
TypeError: emitter(...).ended is not a function

Tracing the issue in the D3 bundle I found out that the move function is called before the emit variable is set:
image

Online debug information:
ContextListener is called first with a TouchStart event
image

Then inside the started function the move function will be called
image

Inside move function the emit variable is undefined, hence throwing the error
image

### PR DESCRIPTION
**Pull request by Fil**
This seems to:
- fix #83
- possibly fix https://github.com/d3/d3-zoom/issues/198
- help with https://github.com/d3/d3-zoom/issues/222

### CODE REVIEW

File: src/brush.js [hunk 1]
```diff
@@ -227,7 +227,7 @@ function brush(dim) {
             function tween(t) {
               state.selection = t === 1 && selection1 === null ? null : i(t);
               redraw.call(that);
-              emit.brush();
+              emit.brush(event);
```
Comment (mbostock) [inline src/brush.js:230]: It doesn’t feel appropriate to use the _event_ here because time has passed—we’re now inside a transition.

**Comment by Fil** [comment]
Hard to tell—can you give a bit more context?

**Comment by lauvigne** [comment]
I have this error on a brush with a filter. 
Theses errors arrive on mobile with touch event, not in my code but in the brush.js.

`this.brush = d3Brush
          .brush()
          .filter(event => event.touches?.length === 2)
          .extent([
            [0, 0],
            [this.context.width, this.height],
          ])
          .on('end', this.brushEnded.bind(this));

this.graphique.append('g').attr('class', 'brush').attr('id', this.indexGraphe).call(this.brush);
`

I will try to do with your sample to reproduce.

Thanks

**Comment by Fil** [comment]
I think maybe .filter(event => event.touches?.length === 2) is filtering
out the events that start the gesture? In that case I can see how you would
get into this situation where the emitter has not been "blessed" with a
touchended property

**Comment by lauvigne** [comment]
OK ! I understood...So It's perhaps a problem with this filter. 
Thanks Fil.

I will investigate on a workaround.

**Pull request merged**